tests/e2e/cleantest_test.go

```diff
@@ -0,0 +1,145 @@
+package e2e
+
+import (
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/odo/tests/e2e/helper"
+)
+
+//  current directory and project (before eny test is run) so it can restored  after all testing is done
+var originalDir string
+var originalProject string
+
+// BeforeSuite is run before whole test suite
+// there can be only one BeforeSuite
+var _ = BeforeSuite(func() {
+	// seed pseudo-random generator
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	// this needs to bedeclared separately
+	// if := is used in the next line it won't save originalDir to global variable
+	var err error
+	originalDir, err = os.Getwd()
+	Expect(err).NotTo(HaveOccurred())
+
+	originalProject = helper.OcCurrentProject()
+})
+
+// AfterSuite is run before whole test suite
+// there can be only one AfterSuite
+var _ = AfterSuite(func() {
+	if originalDir != "" {
+		helper.Chdir(originalDir)
+	}
+	if originalProject != "" {
+		helper.OcSwitchProject(originalProject)
```

Comment:
Could this not cause an overlap? if oc active project is depended on. What happens if one of the parallel runs gets here first and sets a different active project, while another is working with a different project (in a single cluster context)

Reply:
good point. 
It looks like we could use `SynchronizedBeforeSuite` and `SynchronizedAfterSuite`

https://onsi.github.io/ginkgo/#managing-singleton-external-processes-in-parallel-test-suites

Reply:
Fixed by removing those function all together, as I realized that there is a better way to do this